components/calendar/src/chinese.rs

```diff
@@ -384,8 +371,17 @@ impl ChineseBased for Chinese {
 
     const EPOCH: RataDie = CHINESE_EPOCH;
 
-    fn new_chinese_based_date(year: i32, month: u8, day: u8) -> ChineseBasedDateInner<Chinese> {
-        ChineseBasedDateInner(ArithmeticDate::new_unchecked(year, month, day))
+    fn new_chinese_based_date(
+        year: i32,
+        month: u8,
+        day: u8,
+        cache: ChineseBasedCache,
+    ) -> ChineseBasedDateInner<Chinese> {
+        // This can use `new_unchecked` because this function is only ever called from functions which
+        // generate the year, month, and day; therefore, there should never be a situation where
+        // creating this ArithmeticDate would fail, since the same algorithms used to generate the ymd
+        // are also used to check for valid ymd.
```

Comment:
Praise: Thanks for considering this case

Suggestion: this function doesn't do much; consider inlining it, or alternatively add `_unchecked` to the name. It should be self-evident at the call site whether or not our ordinals are validated.

Reply:
Before, this function was needed because it called `Date::try_new_chinese_date` or `Date::try_new_dangi_date`, but since we switched to this returning a `ChineseBasedDateInner` directly, this is no longer needed. So, I removed this function and inlined the contents into `chinese_based_date_from_fixed`